RevenueCat/Scripts/Purchases.cs

```diff
@@ -209,28 +289,119 @@ public void GetProducts(string[] products, GetProductsFunc callback, string type
+    /// iOS only. Initiates a purchase of a <see cref="Package"/>.
+    /// You can get a <c>PromotionalOffer</c> by calling <see cref="GetPromotionalOffer"/>.
+    /// </summary>
+    ///
+    /// <remarks>
+    /// Call this method when a user has decided to purchase a product.
+    /// Only call this in direct response to user input.
+    /// </remarks>
+    ///
+    /// From here the SDK will handle the purchase with <c>StoreKit</c> and call the <c>PurchaseCompletedBlock</c>.
+    ///
+    /// <remarks>
+    /// Note: You do not need to finish the transaction yourself in the completion callback, RevenueCat will
+    /// handle this for you.
+    /// </remarks>
+    ///
+    /// <param name="package"> The <see cref="Package"/> the user intends to purchase.</param>
+    /// <param name="discount"> A <see cref="PromotionalOffer"/> to apply to the purchase.</param>
+    /// <param name="callback"> A <see cref="MakePurchaseCallback"/> completion block that is called when the purchase completes.</param>
+    ///
     public void PurchaseDiscountedPackage(Package package, PromotionalOffer discount, MakePurchaseFunc callback)
     {
         MakePurchaseCallback = callback;
         _wrapper.PurchasePackage(package, discount: discount);
     }
 
+    /// <summary>
+    /// Callback type for methods that return <see cref="CustomerInfo"/>.
+    /// Includes a <see cref="CustomerInfo"/> or an error.
+    /// </summary>
+    /// <param name="customerInfo"> A <see cref="CustomerInfo"/> if the request was successful, null otherwise. </param>
+    /// <param name="error"> An error if the request was not successful, null otherwise. </param>
+    public delegate void CustomerInfoFunc(CustomerInfo customerInfo, Error error);
     private CustomerInfoFunc RestorePurchasesCallback { get; set; }
 
+    /// <summary>
+    /// 
+    /// This method will post all purchases associated with the current Store account to RevenueCat and become
+    /// associated with the current <c>appUserID</c>. If the receipt is being used by an existing user, the current
+    /// <c>appUserID</c> will be aliased together with the <c>appUserID</c> of the existing user.
+    ///  Going forward, either <c>appUserID</c> will be able to reference the same user.
+    /// </summary>
+    ///
+    /// You shouldn't use this method if you have your own account system. In that case "restoration" is provided
+    /// by your app passing the same <c>appUserID</c> used to purchase originally.
+    ///
+    /// - Note: This may force your users to enter their Store password so should only be performed on request of
+    /// the user. Typically with a button in settings or near your purchase UI. Use
+    /// <see cref="SyncPurchases"/> if you need to restore transactions programmatically.
+    ///
+    /// <param name="callback"> A <see cref="CustomerInfoFunc"/> which will contain a <see cref="CustomerInfo"/>
+    /// if restoration was successful, or an error otherwise. </param>
```

Comment:
still a few methods to go here, but this PR was already getting big so I figured I'd split it